python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,539 @@
+                The starting node of the cycle check algorithm.
+
+            inputs: Binding
+                These interfaces are used to connect to each other to build DAG.
+
+            Return
+            ------
+                Return true if there is no cycle in the DAG.
+            """
+            for binding in inputs.values():
+                if start == binding.io_owner:
+                    return False
+                for p in binding.parents:
+                    if not self.check_dag_acyclic(start, p.io_owner.input_bindings.bindings):
+                        return False
+
+            return True
+
+        def connect(self, binding):
+            """Connect the current interface to the destination interface.
+            correct connections as following 1. global input connect to module input,
+            2. module output connect to global output, 3. module output connect to module input
+
+            Parameters
+            ----------
+            binding: Binding
+                The destination of this connection.
+            """
+
+            # Check whether the binding setting is correct or not.
+            if self.io_owner == binding.io_owner:
+                raise RuntimeError(f"Can not bind itself.")
+
+            if not self.is_global_interface() and self.io_type == "input":
+                raise RuntimeError(f"Module can only bind from output interface!")
+
+            if (
+                not self.is_global_interface()
+                and not binding.is_global_interface()
+                and binding.io_type == "output"
+            ):
+                raise RuntimeError(f"Can not bind module output with another module output!")
+
+            if (
+                not self.is_global_interface()
+                and binding.is_global_interface()
+                and binding.io_type == "input"
+            ):
+                raise RuntimeError(f"Can not bind module output with global input!")
+
+            if self.is_global_interface() and self.io_type == "output":
+                raise RuntimeError(f"Global output can not be used as binding start point.")
+
+            if self.is_global_interface() and binding.io_type != "input":
+                raise RuntimeError(f"Global input can only bind with module input.")
+
+            self.bindings.append(binding)
+            if not self.is_global_interface():
+                # Check whether the data types of the source and destination are the same.
+                if (
+                    isinstance(binding.io_owner, PipelineConfig.ModuleWrapper)
+                    and self.data_type != binding.data_type
+                ):
+                    raise RuntimeError(
+                        f"Illegal type (%s vs. %s): binding type is not same!"
+                        % (self.data_type, binding.data_type)
+                    )
+
+                binding.parents.append(self)
+                # Do acyclic check after increase the in-degree.
```

Comment:
increasing

in-degree means `binding.parents`?

Reply:
Fixed.
the in-degree count by the length of binding.parents, changed the comments into
"Do acyclic check after increasing the in-degree of child node by setting current interface as a parent of the child node"